### PR TITLE
[MetricsAdvisor] Remove unnecessary ClientOptions type

### LINKED ISSUE #16369: [MetricsAdvisor] Remove unnecessary ClientOptions type
We currently have:
- Metrics Advisor Client Options
- Metrics Advisor Administration Client Options

These were defined separately for both of our clients, but the service does not differentiate between admin/other operations. We can merge them into the same type: `MetricsAdvisorClientsOptions`.

### PR DESCRIPTION
**Pull request by kinelski**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/16369.

We currently have:

-  Metrics Advisor Client Options
-  Metrics Advisor Administration Client Options

These were defined separately for both of our clients, but the service does not differentiate between admin/other operations. We can merge them into the same type: MetricsAdvisorClientsOptions.

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/CHANGELOG.md [hunk 1]
```diff
@@ -8,17 +8,21 @@
 - Added a public setter to `DataFeed.Options`.
 
 ### Breaking Changes
-- In `MetricsAdvisorClient`, updated `CreateDataFeed` and `CreateDataFeedAsync` to take a whole `DataFeed` object as a parameter.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/CHANGELOG.md:11]: Fixed it. It should be `MetricsAdvisorAdministrationClient`.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/CHANGELOG.md [hunk 2]
```diff
@@ -8,17 +8,21 @@
...
 - In `MetricEnrichedSeriesData`, made elements of `ExpectedValues`, `Periods`, `IsAnomaly`, `LowerBoundaries` and `UpperBoundaries` nullables.
+- Removed `MetricsAdvisorClientOptions` and `MetricsAdvisorAdministrationOptions` and replaced both with `MetricsAdvisorClientsOptions`.
 - Renamed `MetricDimension` to `DataFeedDimension`.
 - Renamed `DataAnomaly` to `DataPointAnomaly`.
 - Renamed `IncidentStatus` to `AnomalyIncidentStatus`.
 - Renamed `AlertingHook`, `EmailHook`, and `WebHook` to `NotificationHook`, `EmailNotificationHook`, and `WebNotificationHook`, respectively.
 - Renamed `TimeMode` to `AlertQueryTimeMode`.
 
+### Key Bug Fixes
+- Fixed a bug in `GetMetricEnrichedSeriesData` sync and async methods where a `NullReferenceException` was thrown if a returned data point had missing data.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/CHANGELOG.md:24]: Missed this in an older PR. Fixed by:
> In `MetricEnrichedSeriesData`, made elements of `ExpectedValues`, `Periods`, `IsAnomaly`, `LowerBoundaries` and `UpperBoundaries` nullables.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Azure.AI.MetricsAdvisor.csproj [hunk 3]
```diff
@@ -3,6 +3,10 @@
     <AssemblyTitle>Microsoft Azure.AI.MetricsAdvisor client library</AssemblyTitle>
     <Version>1.0.0-beta.2</Version>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <NoWarn>
+      $(NoWarn);
+      AZC0007; <!-- https://github.com/Azure/azure-sdk-for-net/issues/16435 -->
+    </NoWarn>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Azure.AI.MetricsAdvisor.csproj:9]: Blocked by https://github.com/Azure/azure-sdk-tools/issues/127#issuecomment-718993646.

I tried suppressing the warning locally, but it still fails when exporting the API.

**Review by maririos — APPROVED** [review]

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs [hunk 4]
```diff
@@ -4,7 +4,7 @@ public partial class MetricsAdvisorClient
     {
         protected MetricsAdvisorClient() { }
         public MetricsAdvisorClient(System.Uri endpoint, Azure.AI.MetricsAdvisor.MetricsAdvisorKeyCredential credential) { }
-        public MetricsAdvisorClient(System.Uri endpoint, Azure.AI.MetricsAdvisor.MetricsAdvisorKeyCredential credential, Azure.AI.MetricsAdvisor.MetricsAdvisorClientOptions options) { }
+        public MetricsAdvisorClient(System.Uri endpoint, Azure.AI.MetricsAdvisor.MetricsAdvisorKeyCredential credential, Azure.AI.MetricsAdvisor.MetricsAdvisorClientsOptions options) { }
```
Comment (christothes) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:7]: The plural *Clients* seems odd to me. Should we just keep the name the same and delete the other one?

Comment (christothes) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:7]: Oh, I see in the linked issue that this is the guideline 😞

**Review by christothes — APPROVED** [review]

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs (hunk 4 above)
Comment (maririos) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:7]: omg I did the same. I was like.. ugh that name looks weird.... looked at guidelines.. ohh nooo :(.
Oh well, sad approve

**Pull request merged**